core/src/packet_deserializer_stage.rs

```diff
@@ -25,6 +30,72 @@ pub struct ReceivePacketResults {
     pub failed_sigverify_count: u64,
 }
 
+pub type DeserializedPacketSender = CrossbeamSender<ReceivePacketResults>;
+
+/// Wrapper for arguments to create a group of packet deserializer threads.
+pub struct PacketDeserializationGroup {
+    pub group_name: &'static str,
+    pub receiver: BankingPacketReceiver,
+    pub sender: DeserializedPacketSender,
+    pub thread_count: usize,
+}
+
+/// Receive packets from sigverify stage, deserialize them, and send to banking stage.
+pub struct PacketDeserializerStage {
+    /// Packet deserialization threads
+    pub thread_handles: Vec<std::thread::JoinHandle<()>>,
+}
+
+impl PacketDeserializerStage {
+    pub(crate) fn new(deserialization_groups: &[PacketDeserializationGroup]) -> Self {
+        let mut thread_handles = Vec::with_capacity(
+            deserialization_groups
+                .iter()
+                .map(|group| group.thread_count)
+                .sum(),
+        );
+
+        for group in deserialization_groups {
+            for thread_index in 0..group.thread_count {
+                let receiver = group.receiver.clone();
+                let sender = group.sender.clone();
+                let packet_deserializer = PacketDeserializer::new(receiver);
+                const RECV_TIMEOUT: Duration = Duration::from_millis(10);
+                thread_handles.push(
+                    Builder::new()
+                        .name(format!("{}{:02}", group.group_name, thread_index))
+                        .spawn(move || {
+                            // Loop while receiving from sigverify
+                            loop {
+                                match packet_deserializer.handle_received_packets(RECV_TIMEOUT) {
```

Comment:
Intentional but not sure if this timeout is optimal here. I kept this in because we'll get some tracer stat aggregation done in the deserializer stage.

There's still a recv timeout on the banking stage channel that receives the deserialized.